.github/workflows/dag-push-staging.yaml

```diff
@@ -0,0 +1,79 @@
+name: Build Packages Staging using dag
+
+on:
+  workflow_dispatch:
+  pull_request:
```

Comment:
This will eventually need to become `pull_request_target` so that the ID token works, but then the workflow definition is only checked out from the repo's head, and not from the PR.